server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -492,6 +514,20 @@ export class AssetService {
           continue;
         }
 
+        // Step 3: Check if any partner owns the asset
+        const partners = (await this.partnerCore.getAll(authUser.id, 'shared-with')).map((item) => item.sharedBy);
+        let owner = false;
+        for (const partnerId of partners) {
+          if ((await this._assetRepository.countByIdAndUser(assetId, partnerId)) == 1) {
+            owner = true;
+            break;
+          }
+        }
+
+        if (owner) {
+          continue;
+        }
+
```

Comment:
Ideally this is added to partnerCore and would just be something like `await this.partnerCore.hasAssetAccess(assetId, authUser.id)`. I think it could be a single query as well. You can join asset to user to partner in a single request.